core/scripts/migrate_archives.py

```diff
@@ -0,0 +1,340 @@
+import argparse
+import itertools
+import os
+from collections import namedtuple
+
+from dbt.config import RuntimeConfig, PROFILES_DIR
+from dbt.adapters.factory import get_adapter
+from dbt.logger import GLOBAL_LOGGER as logger, initialize_logger
+from dbt.clients import system
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--profile',
+        default=None,
+        help='The profile override, if set.'
+    )
+    parser.add_argument(
+        '--target',
+        default=None,
+        help='The target to use within the given profile'
+    )
+    parser.add_argument(
+        '--profiles-dir',
+        default=PROFILES_DIR,
+        help='The directory to look for profiles.yml in'
+    )
+    parser.add_argument(
+        '--debug',
+        action='store_true',
+        help='debug logging'
+    )
+    parser.add_argument(
+        '--apply-files',
+        action='store_true',
+        dest='write_files',
+        help='If set, write .sql files to disk instead of logging them'
+    )
+    parser.add_argument(
+        '--apply-database',
+        action='store_true',
+        dest='migrate_database',
+        help='If set, perform just the database migration'
+    )
+    parser.add_argument(
+        '--apply',
+        action='store_true',
+        help='If set, implies --apply-database --apply-files'
+    )
+    parsed = parser.parse_args()
+    parsed.which = 'migration'
+    if parsed.apply:
+        parsed.write_files = True
+        parsed.migrate_database = True
+    return parsed
+
+
+_UNQUOTE_RENAME_COLUMNS = (
+    ('valid_from', 'dbt_valid_from'),
+    ('valid_to', 'dbt_valid_to'),
+    ('scd_id', 'dbt_scd_id'),
+)
+
+
+ARCHIVE_TEMPLATE = '''
+{{% archive {name} %}}
```

Comment:
should we make this say `snapshot` now? Or do you want to do the rename in dbt first?